executor/show.go

```diff
@@ -1123,19 +1123,57 @@ func appendPartitionInfo(partitionInfo *model.PartitionInfo, buf *bytes.Buffer)
 			}
 		}
 		buf.WriteString(") (\n")
+	} else if partitionInfo.Type == model.PartitionTypeList {
+		if len(partitionInfo.Columns) == 0 {
+			fmt.Fprintf(buf, "\nPARTITION BY %s ( %s ) (\n", partitionInfo.Type.String(), partitionInfo.Expr)
+		} else {
+			colsName := ""
+			for _, col := range partitionInfo.Columns {
+				if len(colsName) > 0 {
```

Comment:
should we use some comment like mysql does?
```
CREATE TABLE `t` (
  `id` int DEFAULT NULL,
  `name` varchar(10) DEFAULT NULL,
  UNIQUE KEY `idx` (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci
/*!50100 PARTITION BY LIST (`id`)
(PARTITION p0 VALUES IN (3,5,6,9,17) ENGINE = InnoDB,
 PARTITION p1 VALUES IN (1,2,10,11,19,20) ENGINE = InnoDB,
 PARTITION p2 VALUES IN (4,12,13,14,18) ENGINE = InnoDB,
 PARTITION p3 VALUES IN (NULL,7,8,15,16) ENGINE = InnoDB) */
```

Reply:
Range/Hash partition doesn't use the comment to, if so, maybe use another PR to change this behavior together?